actionpack/lib/action_dispatch/middleware/debug_exceptions.rb

```diff
@@ -165,7 +165,7 @@ def log_error(request, wrapper)
         exception = wrapper.exception
 
         trace = wrapper.application_trace
-        trace = wrapper.framework_trace if trace.empty?
+        trace = wrapper.framework_trace if trace.empty? && !exception.is_a?(ActionController::RoutingError)
```

Comment:
I'd prefer if we had this conditional on the `wrapper` object instead of here in the middleware, since that class already know about exceptions, possibility making this configurable.